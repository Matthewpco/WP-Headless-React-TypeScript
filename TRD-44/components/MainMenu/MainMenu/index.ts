import { MainMenu, MainMenuProps } from './MainMenu';

export { MainMenu };
export type { MainMenuProps };
