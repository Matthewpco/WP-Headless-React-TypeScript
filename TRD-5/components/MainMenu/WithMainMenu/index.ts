import { WithMainMenu, WithMainMenuProps } from './WithMainMenu';

export { WithMainMenu };
export type { WithMainMenuProps };
