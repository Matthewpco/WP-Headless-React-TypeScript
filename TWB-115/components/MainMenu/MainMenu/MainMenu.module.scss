@import '../../../styles/colors';
@import '../../../styles/media-breakpoints';
@import '../../../styles/theme';

.root {
  border-bottom: solid 1px $athens-gray;
}

.categoryLink {
  color: $black;

  @include ThemeDark {
    color: $ebb;
  }
}

.category:hover .categoryLink {
  color: $punch;
}

.categoryLabel:hover {
  color: $punch;
}

.groupLabel:hover {
  color: $punch;
}

@include mobile_and_tablet {
  .mainMenu {
    display: none;
  }
}

.switchContainer {
  display: none;
}

@include desktop {
  .mainMenu {
    display: flex;
    justify-content: space-between;
  }

  .category {
    position: relative;
    font-family: 'Barlow Condensed', serif;
    font-style: normal;
    font-weight: bold;
    font-size: 16px;
    letter-spacing: 0.25px;
    margin: -1px 0;
    padding: 8px 18px;
  }

  .category:hover {
    background: $hint-of-red-2;
    box-shadow: 0 0 0 1px $athens-gray;
    cursor: pointer;

    @include ThemeDark {
      background: $tuna;
      box-shadow: 0 0 0 1px $tuna;
    }
  }

  .categoryLink {
    display: flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    text-transform: uppercase;
    height: 100%;
  }

  .categoryDropdown {
    display: none;
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
    box-shadow: 0 0 0 1px $athens-gray;
    width: fit-content;
    top: 100%;
    z-index: 2;
  }

  .category:nth-last-child(1) .categoryDropdown,
  .category:nth-last-child(2) .categoryDropdown {
    left: unset;
    right: 0;
    transform: none;
  }

  .category:nth-child(1) .categoryDropdown,
  .category:nth-child(2) .categoryDropdown {
    left: 0;
    transform: none;
  }

  .category:hover .categoryDropdown {
    display: grid;
    grid-template-columns: auto auto auto;
    row-gap: 12px;
    background: $hint-of-red-2;
    padding: 22px 35px;

    @include ThemeDark {
      background: $tuna;
    }
  }

  .categoryDropdown > *:not(:nth-child(3n)):not(:last-child) {
    margin-right: 100px;
  }

  .categoryLabel {
    font-family: Inter, serif;
    font-style: normal;
    font-weight: normal;
    font-size: 13px;
    white-space: nowrap;
    text-decoration: none;
  }

  .itemsGroup {
    display: flex;
    flex-direction: column;
    gap: 14px;
  }

  .groupLabel {
    font-family: Inter, serif;
    font-style: normal;
    font-weight: bold;
    font-size: 13px;
    text-transform: capitalize;
    text-decoration: none;
  }

  .itemsSubGroup {
    display: grid;
    grid-template-columns: auto auto;
    row-gap: 14px;
    column-gap: 40px;
  }
}

