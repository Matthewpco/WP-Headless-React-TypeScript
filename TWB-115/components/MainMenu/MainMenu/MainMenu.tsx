import React from 'react';
import stylesMenu from './MainMenu.module.scss';
import { CoreMenu } from '../CoreMenu/CoreMenu';
import { MainMenuItem } from '../../../graphql';

export type MainMenuProps = {
  data: MainMenuItem[];
};

export const MainMenu = ({ ...props }: MainMenuProps) => (
  <CoreMenu styles={stylesMenu} data={props.data} />
);
