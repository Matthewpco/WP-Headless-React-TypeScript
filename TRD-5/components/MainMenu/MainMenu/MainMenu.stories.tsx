import React from 'react';
import { Story } from '@storybook/react';
import { MainMenu, MainMenuProps } from './MainMenu';

export default { component: MainMenu, title: 'Components/MainMenu' };

const Template: Story<MainMenuProps> = (args) => <MainMenu {...args} />;

const fallbackArgs = {
  data: [
    {
      label: 'Markets',
      url: '#',
      childItems: {
        nodes: [
          {
            label: 'New York',
            url: '#',
          },
          {
            label: 'Chicago',
            url: '#',
          },
          {
            label: 'San Francisco',
            url: '#',
          },
          {
            label: 'Tri-State',
            url: '#',
          },
          {
            label: 'South Florida',
            url: '#',
          },
          {
            label: 'Los Angeles',
            url: '#',
          },
          {
            label: 'Texas',
            url: '#',
          },
        ],
      },
    },
    {
      label: 'News',
      url: '#',
      childItems: {
        nodes: [
          {
            label: 'Residential',
            url: '#',
          },
          {
            label: 'Commercial',
            url: '#',
          },
          {
            label: 'Development',
            url: '#',
          },
          {
            label: 'Landlords',
            url: '#',
          },
          {
            label: 'Politics',
            url: '#',
          },
          {
            label: 'Proptech',
            url: '#',
          },
          {
            label: 'Finance',
            url: '#',
          },
          {
            label: 'Construction',
            url: '#',
          },
        ],
      },
    },
    {
      label: 'Magazine',
      url: '#',
      childItems: {
        nodes: [
          {
            label: 'Current Issue',
            url: '#',
          },
          {
            label: 'Issue archives',
            url: '#',
          },
          {
            label: 'Special Issue Archive',
            url: '#',
          },
        ],
      },
    },
    {
      label: 'TRD PRO',
      url: '#',
    },
    {
      label: 'Events',
      url: '#',
      childItems: {
        nodes: [
          {
            label: 'TRD Events',
            url: '#',
          },
          {
            label: 'Industry Events',
            url: '#',
          },
          {
            label: 'South Florida Showcase',
            url: '#',
          },
        ],
      },
    },
    {
      label: 'Podcast + Video',
      url: '#',
      childItems: {
        nodes: [
          {
            label: 'Podcast',
            url: '#',
            childItems: {
              nodes: [
                {
                  label: 'Deconstruct ',
                  url: '#',
                },
              ],
            },
          },
          {
            label: 'Video',
            url: '#',
            childItems: {
              nodes: [
                {
                  label: 'All Videos',
                  url: '#',
                },
                {
                  label: 'Master Classes',
                  url: '#',
                },
                {
                  label: 'The RE-Interview',
                  url: '#',
                },
                {
                  label: 'Coffee Talk',
                  url: '#',
                },
                {
                  label: 'Real Estate Billionaires',
                  url: '#',
                },
                {
                  label: 'Building Biographies',
                  url: '#',
                },
              ],
            },
          },
        ],
      },
    },
    {
      label: 'Partners',
      url: '#',
      childItems: {
        nodes: [
          {
            label: 'Brand Studio',
            url: '#',
          },
          {
            label: 'Sponsored Content',
            url: '#',
          },
          {
            label: 'Advertise with us',
            url: '#',
          },
        ],
      },
    },
    {
      label: 'Shop',
      url: '#',
    },
  ],
};

export const DefaultMainMenu = Template.bind({});
DefaultMainMenu.args = {
  ...fallbackArgs,
};
