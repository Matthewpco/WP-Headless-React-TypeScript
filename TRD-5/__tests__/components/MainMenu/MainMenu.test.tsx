import React from 'react';
import { render } from '@testing-library/react';
import { MainMenu } from 'components/MainMenu/MainMenu/MainMenu';

describe('MainMenu component', () => {
  it('should render without crash with default props (smoke test)', () => {
    const { container } = render(<MainMenu data={[]} />);
    expect(container).toMatchSnapshot();
  });

  it('should render variant with link-only menu item', () => {
    const id = 'Test id';
    const label = 'Test title';
    const url = 'test';
    const parentId = null;
    const childItems = undefined;

    const { container } = render(
      <MainMenu data={[{ id, label, url, parentId, childItems }]} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render variant with inherit menu items', () => {
    const id = 'Test id';
    const label = 'Test title';
    const url = 'test';
    const parentId = null;
    const childItems = {
      nodes: [
        {
          id: 'test id',
          label: 'Inherit test title',
          url: 'test',
          parentId: null,
        },
      ],
    };

    const { container } = render(
      <MainMenu data={[{ id, label, url, parentId, childItems }]} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render variant with inherit menu items and subitems', () => {
    const id = 'Test id';
    const label = 'Test title';
    const url = 'test';
    const parentId = null;
    const childItems = {
      nodes: [
        {
          id: 'test id',
          label: 'Inherit test title',
          url: 'test-inner-1',
          parentId: null,
          childItems: {
            nodes: [
              {
                id: 'test id',
                label: 'Inherit sutitem test title',
                url: 'test-inner-2',
                parentId: null,
              },
            ],
          },
        },
      ],
    };

    const { container } = render(
      <MainMenu data={[{ id, label, url, parentId, childItems }]} />,
    );
    expect(container).toMatchSnapshot();
  });
});
