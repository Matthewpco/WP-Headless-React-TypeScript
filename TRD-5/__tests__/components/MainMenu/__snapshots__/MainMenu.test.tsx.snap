// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu component should render variant with inherit menu items 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="root container"
    >
      <nav
        class="mainMenu"
      >
        <div
          class="category"
        >
          <a
            class="categoryLink"
            draggable="false"
            href="/#"
          >
            Test title
          </a>
          <div
            class="categoryDropdown"
          >
            <a
              class="categoryLabel"
              draggable="false"
              href="/test"
            >
              Inherit test title
            </a>
          </div>
        </div>
      </nav>
      <div
        class="switchContainer"
      >
        <button
          class="root rootActive"
          type="button"
        >
          <div
            class="highlight"
          />
          SvgMock
          SvgMock
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`MainMenu component should render variant with inherit menu items and subitems 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="root container"
    >
      <nav
        class="mainMenu"
      >
        <div
          class="category"
        >
          <a
            class="categoryLink"
            draggable="false"
            href="/#"
          >
            Test title
          </a>
          <div
            class="categoryDropdown"
          >
            <div
              class="itemsGroup"
            >
              <a
                class="groupLabel"
                draggable="false"
                href="/test-inner-1"
              >
                Inherit test title
              </a>
              <div
                class="itemsSubGroup"
              >
                <a
                  class="categoryLabel"
                  draggable="false"
                  href="/test-inner-2"
                >
                  Inherit sutitem test title
                </a>
              </div>
            </div>
          </div>
        </div>
      </nav>
      <div
        class="switchContainer"
      >
        <button
          class="root rootActive"
          type="button"
        >
          <div
            class="highlight"
          />
          SvgMock
          SvgMock
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`MainMenu component should render variant with link-only menu item 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="root container"
    >
      <nav
        class="mainMenu"
      >
        <div
          class="category"
        >
          <a
            class="categoryLink"
            draggable="false"
            href="/test"
          >
            Test title
          </a>
        </div>
      </nav>
      <div
        class="switchContainer"
      >
        <button
          class="root rootActive"
          type="button"
        >
          <div
            class="highlight"
          />
          SvgMock
          SvgMock
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`MainMenu component should render without crash with default props (smoke test) 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="root container"
    >
      <nav
        class="mainMenu"
      />
      <div
        class="switchContainer"
      >
        <button
          class="root rootActive"
          type="button"
        >
          <div
            class="highlight"
          />
          SvgMock
          SvgMock
        </button>
      </div>
    </div>
  </div>
</div>
`;
